Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs

```diff
@@ -923,6 +928,10 @@ private void SpawnOperatives(int spawnCount, List<IPlayerSession> sessions, bool
                 _mindSystem.SetUserId(newMind, session.UserId);
                 _roles.MindAddRole(newMind, new NukeopsRoleComponent { PrototypeId = spawnDetails.Role });
 
+                // Automatically de-admin players who are being made nukeops
+                if (_cfg.GetCVar(CCVars.AdminDeadminOnJoin) && _adminManager.IsAdmin(session))
+                    _adminManager.DeAdmin(session);
+
```

Comment:
Manually having to deadmin seems much more prone to bugs than just doing this in a central location and checking if they get a role attached?

Reply:
Not sure exactly how that would work, to be honest. If you have ideas I'm all for it, but given this is (essentially) how the cvar worked already it seems easiest to just do this.